Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-details/ContestDetailsPage.tsx

```diff
@@ -0,0 +1,158 @@
+import ContestButton from '../../../components/contests/contest-button/ContestButton';
+import Button from '../../../components/guidelines/buttons/Button';
+import Heading, { HeadingType } from '../../../components/guidelines/headings/Heading';
+import SpinningLoader from '../../../components/guidelines/spinning-loader/SpinningLoader';
+import useTheme from '../../../hooks/use-theme';
+import { setContestDetails } from '../../../redux/features/contestsSlice';
+import { useGetContestByIdQuery } from '../../../redux/services/contestsService';
+import { useAppDispatch, useAppSelector } from '../../../redux/store';
+import { flexCenterObjectStyles } from '../../../utils/object-utils';
+import { getContestTypeResultsUrl } from '../../../utils/urls';
+
+import styles from './ContestDetailsPage.module.scss';
+
+const ContestDetailsPage = () => {
+    const dispatch = useAppDispatch();
+    const navigate = useNavigate();
+    const { contestId } = useParams();
+    const { internalUser: user } = useAppSelector((state) => state.authorization);
+    const { themeColors, getColorClassName } = useTheme();
+    const { contestDetails, selectedCategory } = useAppSelector((state) => state.contests);
+    const { data, isLoading, error } = useGetContestByIdQuery({ id: Number(contestId) });
+
+    const textColorClassName = getColorClassName(themeColors.textColor);
+    const {
+        id,
+        name,
+        allowedSubmissionTypes,
+        description,
+        problems,
+        competeParticipantsCount,
+        practiceParticipantsCount,
+        canBeCompeted,
+        canBePracticed,
+    } = data ?? {};
+
+    useEffect(() => {
+        if (contestDetails?.id !== data?.id) {
+            dispatch(setContestDetails({ contest: data ?? null }));
+        }
+    }, [ data, contestDetails, dispatch ]);
+
+    const renderAllowedLanguages = () => allowedSubmissionTypes?.map((allowedSubmissionType) => (
+        <>
+            <Link
+              key={`contest-sub-strat-btn-${allowedSubmissionType.id}`}
+              className={styles.allowedLanguageLink}
+              to={`/contests?category=${selectedCategory?.id}&strategy=${allowedSubmissionType.id}`}
+            >
+                {allowedSubmissionType.name}
+            </Link>
+            {' | '}
+        </>
+    ));
+
+    const renderProblemsNames = () => {
+        if (!problems) {
+            return 'The problems for this contest are not public.';
+        }
+        return problems.map((problem) => (
+            <div className={styles.problemNameItem}>
+                {problem.name}
+            </div>
+        ));
+    };
+
+    const renderAdministrationButtons = () => (
+        <div>
+            <Button onClick={() => navigate(`/administration-new/contests/${id}`)}>Edit</Button>
+            <Button className={styles.adminBtn} onClick={() => navigate(`/contests/${id}/compete/results/full`)}>Full Results</Button>
+        </div>
+    );
+
```

Comment:
The routes should be moved to a separate file, because if they are used on many places this will cause problems for the future. Even if they are used in one place this doesn't guarantee that for the future they will not be reused elsewhere.

Reply:
done